src/models/forecasts.jl

```diff
@@ -2,13 +2,44 @@ abstract type
     Forecast
 end
 
+"""
+    Deterministic
+        A deterministic forecast for a particular data field in a PowerSystemDevice.
+
+"""
 struct Deterministic <: Forecast
-    device::PowerSystemDevice
-    horizon::Int
-    resolution::Dates.Period
-    interval::Dates.Period
-    initialtime::Dates.DateTime
-    data::Dict{Any,Dict{Int,TimeSeries.TimeArray}}
+    device::PowerSystemDevice       # device
+    field::Symbol                   # field of device
+    resolution::Dates.Period        # resolution
+    initialtime::Dates.DateTime     # forecast availability time
+    data::TimeSeries.TimeArray      # TimeStamp - scalingfactor
+
+    function Deterministic(device::PowerSystemDevice,
+                            field::Symbol,
+                            resolution::Dates.Period,
+                            initialtime::Dates.DateTime,
+                            data::TimeSeries.TimeArray; kwargs...)
+        runchecks = in(:runchecks, keys(kwargs)) ? kwargs[:runchecks] : true
```

Comment:
This idiom commonly exists as well.  Suggest moving toward this:

runchecks = get(kwargs, :runchecks, true)